app/services/registration/policy_engine.rb

```diff
@@ -0,0 +1,54 @@
+module Registration
+  class PolicyEngine
+    Result = Struct.new(:pass, :failed_policy, :trace, keyword_init: true)
+
+    def initialize(campaign)
+      @campaign = campaign
+    end
+
+    def eligible?(user, phase: :registration)
+      trace = []
+
+      policies_for_phase(phase).each do |policy|
+        outcome = policy.evaluate(user)
+        trace << {
+          policy_id: policy.id,
+          kind: policy.kind,
+          phase: policy.phase,
+          outcome: outcome
+        }
+
+        unless outcome[:pass]
+          return Result.new(
+            pass: false,
+            failed_policy: policy,
+            trace: trace
+          )
+        end
+      end
+
+      Result.new(pass: true, failed_policy: nil, trace: trace)
+    end
+
+    def full_trace_for(user, phase: :registration)
```

Comment:
Where is this method needed? I don't find any code occurrence, nor an occurence in the architecture book.

Reply:
I removed the method and added the link to the architecture book.